samples/streamFileDownload.js

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This application demonstrates how to perform basic operations on files with
+ * the Google Cloud Storage API.
+ *
+ * For more information, see the README.md under /storage and the documentation
+ * at https://cloud.google.com/storage/docs.
+ */
+const path = require('path');
+const cwd = path.join(__dirname, '..');
+const fs = require('fs');
+
+function main(
+  bucketName = 'my-bucket',
+  fileName = 'test.txt',
+  destFileName = path.join(cwd, 'downloaded.txt')
+) {
+  // [START storage_stream_file_download]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // The ID of your GCS bucket
+  // const bucketName = 'your-unique-bucket-name';
+
+  // The ID of your GCS file
+  // const fileName = 'your-file-name';
+
+  // The path to which the file should be downloaded
+  // const destFileName = '/local/path/to/file.txt';
+
+  // Imports the Google Cloud client library
+  const {Storage} = require('@google-cloud/storage');
+
+  // Creates a client
+  const storage = new Storage();
+
+  async function streamFileDownload() {
+    // The example below demonstrates how we can reference a remote file, then
+    // pipe its contents to a local file. This is effectively creating a local
+    // backup of your remote data.
+    await storage
+      .bucket(bucketName)
+      .file(fileName)
+      .createReadStream()
```

Comment:
I would maybe separate out a variable for the stream created at this step, and add a comment that you can do whatever you want to do with the stream (not just pipe it into a file necessarily).

Reply:
Do you think that might add too much complexity to the sample? There is tons of nodejs documentation about using streaming and things to do with streams. This sample just aims to show how to use streams with storage.

Reply:
I think the main difference between a streaming download and a standard download is simply the fact that that the stream is something that can be used in different ways, right? Otherwise this will just end up looking identical to the standard download sample in some languages (e.g. Go).

Reply:
The docs page says "Cloud Storage supports streaming transfers, which allow you to stream data to and from your Cloud Storage account without requiring that the data first be saved to a file."

This makes me think it might be better to have the sample stream the contents of the file to stdout or something like that.

Reply:
I added the 'finish' function so it's a little different from the normal download sample. it now shows a stream function and how it can be used. i'm slightly opposed to separating out the stream into its own variable because that's not how we have any of the stream examples currently in our repo. This sample is consistent with all our other documentation.